synapse/handlers/appservice.py

```diff
@@ -223,7 +222,7 @@ async def _handle_typing(self, service: ApplicationService, new_token: int):
             # For performance reasons, we don't persist the previous
             # token in the DB and instead fetch the latest typing information
             # for appservices.
-            from_key=new_token - 1,
+            from_key=int(new_token) - 1,
```

Comment:
Why this change? Isn't `new_token` always an int? (According to the type hints...)

Reply:
argh, that was me trying to combat `Union` types before I realized I needed to do my type checking earlier.